tracer/test/Datadog.Trace.Tools.Runner.Tests/CodeCoverageAdapterInjectorTests.cs

```diff
@@ -0,0 +1,241 @@
+// <copyright file="CodeCoverageAdapterInjectorTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System.Collections.Generic;
+using Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.DotnetTest;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Util;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Xunit;
+
+namespace Datadog.Trace.Tools.Runner.Tests;
+
+public class CodeCoverageAdapterInjectorTests
+{
+    public static IEnumerable<object[]> DotnetTestData()
+    {
+        string[] emptyStringArray = [string.Empty];
+        yield return [emptyStringArray, emptyStringArray.Concat(["-property:VSTestCollect=\"DatadogCoverage\"", "-property:VSTestTestAdapterPath=\"TempPath\""])];
+
+        string[] spaceStringArray = [" "];
+        yield return [spaceStringArray, spaceStringArray.Concat(["-property:VSTestCollect=\"DatadogCoverage\"", "-property:VSTestTestAdapterPath=\"TempPath\""])];
+
+        string[] example1 =
```

Comment:
because the injection point on the `dotnet test` arguments is done after the parsing of the `--` all arguments after are moved into the `-property:VSTestCLIRunSettings` argument that we are testing.

You will see that it's different in the case of the `vstest.console` argument injection, in that case we can inject before the parsing, so we have tests for the `--` case there.